### PR TITLE
fix: type error and border

### LINKED ISSUE #6489: [CAL-792] routing forms dropdown fix
* [X] 1. @PeerRich in the availability design there was no trash icon for delete option. Should I add it?
* [ ] 2. Will fix this in routing forms

*Originally posted by @Udit-takkar in* [*https://github.com/calcom/cal.com/issues/6440#issuecomment-1382651927*](https://github.com/calcom/cal.com/issues/6440#issuecomment-1382651927)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-792](https://linear.app/calcom/issue/CAL-792/routing-forms-dropdown-fix)</sub>

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/6489

https://user-images.githubusercontent.com/53316345/212867648-bf337957-d63c-4a70-8db0-9c310118e007.mov

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: packages/ui/components/form/dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -102,7 +102,7 @@ DropdownMenuRadioItem.displayName = "DropdownMenuRadioItem";
 
 type DropdownItemProps = {
   children: React.ReactNode;
-  color?: ButtonProps["color"];
```
Comment (Udit-takkar) [inline packages/ui/components/form/dropdown/Dropdown.tsx:105]: there was some type error using this so used ButtonColor instead

File: packages/app-store/ee/routing-forms/components/FormActions.tsx [hunk 2]
```diff
@@ -447,7 +447,10 @@ export const FormAction = forwardRef(function FormAction<T extends typeof Button
   }
   return (
     <DropdownMenuItem>
-      <Component ref={forwardedRef} {...actionProps}>
+      <Component
+        ref={forwardedRef}
+        {...actionProps}
+        className={classNames(props.className, "transition-none")}>
```
Comment (Udit-takkar) [inline packages/app-store/ee/routing-forms/components/FormActions.tsx:453]: transition-colors in button was the cause of border flicker. ( this took a lot of time)

File: packages/ui/components/form/dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { ComponentProps, forwardRef } from "react";
 
 import { classNames } from "@calcom/lib";
 import { SVGComponent } from "@calcom/types/SVGComponent";
-import { ButtonProps } from "@calcom/ui";
+import { ButtonColor } from "@calcom/ui";
```
Comment (sean-brydon) [inline packages/ui/components/form/dropdown/Dropdown.tsx:8]:
```suggestion
import type { ButtonColor } from "@calcom/ui";
```

**Review by sean-brydon — APPROVED** [review]
LGTM :) We have a weird margin-y on this it looks like. Don't think its coming from here tho

 image

**Comment by PeerRich** [comment]
> LGTM :) We have a weird margin-y on this it looks like. Don't think its coming from here tho
> 
> image

according to @Jaibles thats supposed to

**Comment by Jaibles** [comment]
@sean-brydon because of gestalts law of proximity if you want the technical answer 😂

If you think it's **really** ugly I've proposed an alternative option in the loom here
https://www.loom.com/share/faef67d3d9e0464aba64c4fe8ea949d9

**Pull request merged**